backend/tracim_backend/lib/core/content.py

```diff
@@ -2112,34 +2165,21 @@ def get_all_types(self) -> typing.List[ContentType]:
 
         return content_types
 
-    # TODO - G.M - 2018-07-24 - [Cleanup] Is this method already needed ?
-    def exclude_unavailable(self, contents: typing.List[Content]) -> typing.List[Content]:
-        """
-        Update and return list with content under archived/deleted removed.
-        :param contents: List of contents to parse
-        """
-        for content in contents[:]:
-            if self.content_under_deleted(content) or self.content_under_archived(content):
-                contents.remove(content)
-        return contents
-
-    # TODO - G.M - 2018-07-24 - [Cleanup] Is this method already needed ?
-    def content_under_deleted(self, content: Content) -> bool:
+    def get_deleted_parent_id(self, content: Content) -> typing.Optional[int]:
         if content.parent:
             if content.parent.is_deleted:
-                return True
+                return content.parent_id
             if content.parent.parent:
-                return self.content_under_deleted(content.parent)
-        return False
+                return self.get_deleted_parent_id(content.parent)
+        return None
```

Comment:
For information, for a function with name `get_xxxx` better design is to raise an exception instead return None if expected result cannot be found.

Reply:
see #1875